apps/src/sites/studio/pages/googleBlocklyWrapper.js

```diff
@@ -0,0 +1,191 @@
+
+  blocklyWrapper.wrapSettableProperty('assetUrl');
+  blocklyWrapper.wrapSettableProperty('behaviorEditor');
+  blocklyWrapper.wrapSettableProperty('BROKEN_CONTROL_POINTS');
+  blocklyWrapper.wrapSettableProperty('BUMP_UNCONNECTED');
+  blocklyWrapper.wrapSettableProperty('HSV_SATURATION');
+  blocklyWrapper.wrapSettableProperty('JavaScript');
+  blocklyWrapper.wrapSettableProperty('readOnly');
+  blocklyWrapper.wrapSettableProperty('showUnusedBlocks');
+  blocklyWrapper.wrapSettableProperty('SNAP_RADIUS');
+  blocklyWrapper.wrapSettableProperty('typeHints');
+  blocklyWrapper.wrapSettableProperty('valueTypeTabShapeMap');
+
+  blocklyWrapper.getGenerator = function() {
+    return this.JavaScript;
+  };
+  blocklyWrapper.findEmptyContainerBlock = function() {}; // TODO
+  blocklyWrapper.BlockSpace = {};
+  blocklyWrapper.BlockSpace.EVENTS = {};
+  blocklyWrapper.BlockSpace.EVENTS.MAIN_BLOCK_SPACE_CREATED =
+    'mainBlockSpaceCreated';
+  blocklyWrapper.BlockSpace.EVENTS.EVENT_BLOCKS_IMPORTED = 'blocksImported';
+  blocklyWrapper.BlockSpace.EVENTS.BLOCK_SPACE_CHANGE = 'blockSpaceChange';
+  blocklyWrapper.BlockSpace.EVENTS.BLOCK_SPACE_SCROLLED = 'blockSpaceScrolled';
+  blocklyWrapper.BlockSpace.EVENTS.RUN_BUTTON_CLICKED = 'runButtonClicked';
+  blocklyWrapper.BlockSpace.onMainBlockSpaceCreated = () => {}; // TODO
+  blocklyWrapper.BlockSpace.createReadOnlyBlockSpace = () => {}; // TODO
+
+  blocklyWrapper.Block.prototype.getTitleValue =
+    blocklyWrapper.Block.prototype.getFieldValue;
+  blocklyWrapper.Block.prototype.setHSV = function(h, s, v) {
+    return this.setColour(h);
+  };
+
+  blocklyWrapper.Input.prototype.appendTitle = function(a, b) {
+    return this.appendField(a, b);
+  };
+
+  Object.defineProperty(blocklyWrapper, 'mainBlockSpace', {
+    get: function() {
+      return this.blockly_.mainWorkspace;
+    }
+  });
+  Object.defineProperty(blocklyWrapper, 'mainBlockSpaceEditor', {
+    get: function() {
+      return this.blockly_.mainWorkspace;
+    }
+  });
+  Object.defineProperty(blocklyWrapper, 'SVG_NS', {
+    get: function() {
+      return this.blockly_.utils.dom.SVG_NS;
+    }
+  });
+
+  blocklyWrapper.Workspace.prototype.getToolboxWidth = function() {
+    return blocklyWrapper.mainBlockSpace.getMetrics().toolboxWidth;
+  };
+
+  blocklyWrapper.Xml.domToBlockSpace = blocklyWrapper.Xml.domToWorkspace;
+  blocklyWrapper.Xml.blockSpaceToDom = blocklyWrapper.Xml.workspaceToDom;
+  blocklyWrapper.Workspace.prototype.addUnusedBlocksHelpListener = () => {}; // TODO
+  blocklyWrapper.Workspace.prototype.getAllUsedBlocks =
+    blocklyWrapper.Workspace.prototype.getAllBlocks; // TODO
+  blocklyWrapper.Workspace.prototype.isReadOnly = () => false; // TODO
+  blocklyWrapper.Workspace.prototype.setEnableToolbox = () => {}; // TODO
+  blocklyWrapper.Workspace.prototype.blockSpaceEditor = {
+    blockLimits: {
+      blockLimitExceeded: () => false, // TODO
+      getLimit: () => {} // TODO
+    }
+  };
```

Comment:
Perhaps can be reduced to one command?

Reply:
So it's definitely possible that there's a way to do this that I just don't know about, but I'm not sure that will work for the Workspace because we need to set all of these properties on the prototype.
So like, we can do 
```
blocklyWrapper.Xml = {
  ...blocklyWrapper.Xml,
  // our custom properties here
}
```
for the Xml, but for Workspace it would need to look like
```
blocklyWrapper.Workspace.prototype = {
  ...blocklyWrapper.Workspace.prototype,
  // custom properties
}
```
which doesn't seem to work, but also re-assigning the prototype entirely seems potentially scary/unwise. 

What do you think @breville ?

Reply:
Oh, very interesting.  It's [apparently possible](https://stackoverflow.com/questions/49111307/typescript-set-object-prototype-with-spread-operator) to use `Object.assign` and `Object.create` together, though I don't feel strongly about changing it from what you already have.